apiary-receivers/apiary-receiver-common/src/main/java/com/expedia/apiary/extensions/receiver/common/messaging/MessageReader.java

```diff
@@ -18,13 +18,13 @@
 import java.io.Closeable;
 import java.util.Optional;
 
-import com.expedia.apiary.extensions.receiver.common.event.ListenerEvent;
-
 /**
  * A {@code MessageReader} is in charge of retrieving events from the messaging infrastructure.
  */
 public interface MessageReader extends Closeable {
 
-  Optional<ListenerEvent> read();
+  Optional<MessageEvent> read();
+
+  void delete(String messageId);
```

Comment:
Let's use the `MessageEvent` here and let the implementation deal with message ids or whatever else it needs in order to do the delete.

Reply:
Done :)